src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -227,13 +226,25 @@ export function updateLayeredMaterialNodeImagery(context, layer, node) {
         return;
     }
 
+    const failureParams = node.layerUpdateState[layer.id].failureParams;
     const currentLevel = node.material.getColorLayerLevelById(layer.id);
-    const zoom = node.getCoordsForLayer(layer)[0].zoom || node.level;
-    const targetLevel = chooseNextLevelToFetch(layer.updateStrategy.type, node, zoom, currentLevel, layer);
+    const bestLevel = node.getCoordsForLayer(layer)[0].zoom || node.level;
+    // if there's failureParams, compute the next level to try by dichotomic calculation
+    // dichotomy between the current level and last failed level
+    const nextLevelToTry = failureParams ? Math.ceil((currentLevel + failureParams.targetLevel) / 2) : undefined;
+    const targetLevel = nextLevelToTry || chooseNextLevelToFetch(layer.updateStrategy.type, node, bestLevel, currentLevel, layer);
```

Comment:
Now that this code has been simplified, it seems that it would be better in `chooseNextLevelToFetch`...

This way the code would be shared between imagery and elevation and the update process code would be simpler.
You may want to enrich the `LayerUpdateState` object to remember the N previous failure, so that it can decide if it wants to continue (using dichotomy) or stop.

Reply:
in first step, I move the code in chooseNextLevelToFetch

Reply:
@peppsac for the first step, WDYT?